test/e2e/network/network_policy.go

```diff
@@ -352,12 +486,552 @@ var _ = SIGDescribe("NetworkPolicy", func() {
+					},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+					Ingress:     []networkingv1.NetworkPolicyIngressRule{{}},
+				},
+			}
+
+			policyAllowAll, err = f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyAllowAll)
+			Expect(err).NotTo(HaveOccurred())
+			defer cleanupNetworkPolicy(f, policyAllowAll)
+
+			// client-a should be able to contact server as the policy is to allow all
+			By("Creating client-a which should be able to contact the server.", func() {
+				testCanConnect(f, f.Namespace, "client-a", service, 80)
+			})
+		})
+
+		It("should enforce policies, where allow-all policy take precedence over other egress policy, based on PodSelector [Feature:NetworkPolicy]", func() {
+			By("Creating a network policy for the server which allows traffic only to client-b.")
+			policyDenyClientA := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: f.Namespace.Name,
+					Name:      "allow-from-client-b-pod-selector",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Apply this policy to the Server
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"pod-base-name": podServerLabelSelector,
+						},
+					},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+					// Allow traffic only to "client-b"
+					Egress: []networkingv1.NetworkPolicyEgressRule{{
+						To: []networkingv1.NetworkPolicyPeer{{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"pod-base-name": "client-b",
+								},
+							},
+						}},
+					}},
+				},
+			}
+
+			policyDenyClientA, err := f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyDenyClientA)
+			Expect(err).NotTo(HaveOccurred())
+			defer cleanupNetworkPolicy(f, policyDenyClientA)
+
+			// client-a should not be able to contact server as the policy is to allow only client-b
+			By("Creating client-a which should not be able to contact the server.", func() {
+				testCannotConnect(f, f.Namespace, "client-a", service, 80)
+			})
+
+			By("Creating a network policy for the server which allows traffic from all clients.")
+			policyAllowAll := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: f.Namespace.Name,
+					Name:      "allow-all",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Apply this policy to allow all pods
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{},
+					},
+					PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+					Ingress:     []networkingv1.NetworkPolicyIngressRule{{}},
+				},
+			}
+
+			policyAllowAll, err = f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policyAllowAll)
+			Expect(err).NotTo(HaveOccurred())
+			defer cleanupNetworkPolicy(f, policyAllowAll)
+
+			// client-a should be able to contact server as the policy is to allow all
```

Comment:
This is wrong; ingress and egress are separate things. The "allow-all" ingress policy says that all pods will *accept* traffic from all other pods, but the egress policy above is still in effect and says that "client-a" can only *send* traffic to "client-b". (In other words, at this point, the server would be happy to accept traffic from client-a if client-a could send that traffic, but client-a cannot send that traffic, so the server never gets the opportunity to accept it.)

Did you test these new tests against some network plugin? If so, and that plugin passed this test, then that plugin is broken. (In which case, hooray for having more tests!)

Reply:
Acknowledged and changed the test.

Reply:
@danwinship has this comment be addressed now? 

I think it has, just looking at the current code (there isn't any mention of ingress in this test any more and it seems to be doing the right thing).